src/Compilers/CSharp/Test/Semantic/Semantics/PatternMatchingTestBase.cs

```diff
@@ -147,6 +157,17 @@ protected static void VerifyModelForDeclarationOrVarPatternDuplicateInSameScope(
             }
         }
 
+        protected static void VerifyModelForDeclarationOrVarPatternDuplicateInSameScope(SemanticModel model, VariableDeclaratorSyntax declarator)
+        {
+            var symbol = model.GetDeclaredSymbol(declarator);
+            Assert.Equal(declarator.Identifier.ValueText, symbol.Name);
+            Assert.Equal(declarator, symbol.DeclaringSyntaxReferences.Single().GetSyntax());
+            Assert.Equal(LocalDeclarationKind.RegularVariable, ((LocalSymbol)symbol).DeclarationKind);
```

Comment:
>Assert.Equal(LocalDeclarationKind.RegularVariable, ((LocalSymbol)symbol).DeclarationKind); [](start = 12, length = 90)

It doesn't look like we are verifying model for a pattern variable as the name of the method suggests. Consider renaming the method to reflect its purpose more accurately.